File: CustomAnimDemoApplication/app/src/main/java/org/lynxz/customanimdemoapplication/BaseActivity.java
package org.lynxz.customanimdemoapplication;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

/**
 * Created by zxz on 2016/4/26.
 */
public class BaseActivity extends Activity {


    public <T extends View> T findView(int id) {
        return (T) findViewById(id);
    }

    public void showToast(String msg) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }
}

File: RecycleViewDemo/app/src/main/java/org/lynxz/recycleviewdemo/activity/BaseActivity.kt
package org.lynxz.recycleviewdemo.activity

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

/**
 * Created by lynxz on 02/02/2018.
 * 博客: https://juejin.im/user/5812c2b0570c3500605a15ff
 */
abstract class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutRes())
        afterCreate()
    }

    /**
     * 获取布局id
     * */
    @LayoutRes
    abstract fun getLayoutRes(): Int

    abstract fun afterCreate()
}